From 8a93a04f5179a242b8c9a69504040a5286e2079b Mon Sep 17 00:00:00 2001
Message-Id: <8a93a04f5179a242b8c9a69504040a5286e2079b.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 19 Nov 2014 09:19:25 +0530
Subject: [PATCH 17/75] MUST_REBASE [VPG]: drm/i915/adf: Respect ADF kernel
 paramter before initilizing ADF driver

Exporting a i915_adf_driver_initilize function to help adf driver to
decide if ADF driver should be initilized or not for a given platform

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need
to work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I6751ca0cee22a4e830094aabb086a11d86b3638e
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |   13 +++++++++++++
 include/drm/i915_adf.h                  |    1 +
 2 files changed, 14 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index bb5c61f..b5d10c3 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -66,6 +66,19 @@ void set_adf_ready(void)
 	g_adf_ready = true;
 }
 
+/**
+ * i915_adf_driver_initialize - Adf driver calls this function to check if
+ * kernel paramter for ADF Enable is set or notice
+ */
+int i915_adf_driver_initialize(void)
+{
+	if (!i915_adf_dev)
+		return 0;
+
+	return i915.enable_intel_adf;
+}
+EXPORT_SYMBOL(i915_adf_driver_initialize);
+
 struct pci_dev *i915_adf_get_pci_dev(void)
 {
 	if (!i915_adf_dev)
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 40bc75c..da9fb64 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -55,6 +55,7 @@ extern volatile bool g_adf_ready;
 extern int intel_adf_context_on_event(void);
 extern void intel_adf_display_rpm_get(void);
 extern void intel_adf_display_rpm_put(void);
+extern int i915_adf_driver_initialize(void);
 #endif
 
 
-- 
1.7.9.5

